Mage/src/main/java/mage/abilities/effects/common/DraftFromSpellbookThenExileCastThisTurnEffect.java

```diff
@@ -0,0 +1,82 @@
+package mage.abilities.effects.common;
+
+import mage.abilities.Ability;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.asthought.PlayFromNotOwnHandZoneTargetEffect;
+import mage.cards.Card;
+import mage.cards.repository.CardCriteria;
+import mage.cards.repository.CardInfo;
+import mage.cards.repository.CardRepository;
+import mage.choices.Choice;
+import mage.choices.ChoiceHintType;
+import mage.choices.ChoiceImpl;
+import mage.constants.Duration;
+import mage.constants.Outcome;
+import mage.constants.TargetController;
+import mage.game.Game;
+import mage.players.Player;
+import mage.util.RandomUtil;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * @author Sidorovich77
+ */
+public class DraftFromSpellbookThenExileCastThisTurnEffect extends OneShotEffect {
```

Comment:
Instead copy paste full code from existing effect -- extends `DraftFromSpellbookEffect` and override apply method like: "if super.apply then exile/cast"

Reply:
Maybe should be a method in Player class `Set<Card> draftFromSpellbook(List<String> spellbook, PutCards putCards)` to handle the functionality